smart-contracts/contracts/Unlock.sol

```diff
@@ -203,14 +217,11 @@ contract Unlock is
     if(_value > 0) {
       uint valueInETH;
       address tokenAddress = IPublicLock(msg.sender).tokenAddress();
-      if(tokenAddress != address(0)) {
-        // If priced in an ERC-20 token, find the supported uniswap exchange
-        IUniswapExchange exchange = uniswapExchanges[tokenAddress];
-        if(address(exchange) != address(0)) {
-          valueInETH = exchange.getTokenToEthInputPrice(_value);
-        } else {
-          // If the token type is not supported, assume 0 value
-          valueInETH = 0;
+      if(tokenAddress != address(0) && tokenAddress != weth) {
```

Comment:
Does that mean we will nto account for locks in Weth? I think we should also use them?

Reply:
UniswapV2 doesn't support ETH under the hood. So we are converting to weth, and if the tokenAddress is ETH or WETH then we skip Uniswap and use the value directly in the else block below.

Reply:
I think in that case we should just set `valueInETH = _value`

Reply:
Correct, that's what we do in the else block below